frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/utils.ts

```diff
@@ -265,16 +265,25 @@ const removeListRunAfters = (task: PipelineTask, listIds: string[]): PipelineTas
   return task;
 };
 
-const removeEmptyDefaultParams = (task: PipelineTask): PipelineTask => {
+export const removeEmptyFormFields = (task: PipelineTask): PipelineTask => {
+  let output = task;
+  // Since we can submit, this param has a default; check for empty values and remove
   if (task.params?.length > 0) {
-    // Since we can submit, this param has a default; check for empty values and remove
-    return {
-      ...task,
-      params: task.params.filter((param) => !!param.value),
-    };
+    const params = task.params?.filter((param) => !!param.value);
+    output = { ...output, params };
   }
-
-  return task;
+  // Drop input/output resources which are not linked to an (optional) resource.
+  if (task.resources?.inputs?.length > 0 || task.resources?.outputs?.length > 0) {
+    const inputs = task.resources?.inputs?.filter((resource) => resource.resource);
+    const outputs = task.resources?.outputs?.filter((resource) => resource.resource);
+    output = { ...output, resources: { ...output.resources, inputs, outputs } };
+  }
+  // Drop workspaces which are not linked to an (optional) workspace.
+  if (task.workspaces?.length > 0) {
+    const workspaces = task.workspaces?.filter((workspace) => workspace.workspace);
+    output = { ...output, workspaces };
+  }
+  return output;
```

Comment:
I think you can revert this change based on my other comments.

Reply:
We need this also if the user selects "No image resource" or "No workspace". This is required because the form task data contains a list of all configurable resources and workspaces. If the user does not select an optional field, this entries must be removed before the API was called. Or did I missed another comment?